src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt

```diff
@@ -0,0 +1,367 @@
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.GroupBatch
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.Message
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.TransportGroupBatchRouter.Companion.TRANSPORT_GROUP_ATTRIBUTE
+import com.exactpro.th2.common.utils.event.transport.toProto
+import com.exactpro.th2.common.utils.message.transport.toGroup
+import com.exactpro.th2.http.client.annotations.IntegrationTest
+import com.exactpro.th2.test.annotations.Th2AppFactory
+import com.exactpro.th2.test.annotations.Th2IntegrationTest
+import com.exactpro.th2.test.annotations.Th2TestFactory
+import com.exactpro.th2.test.queue.CollectorMessageListener
+import com.exactpro.th2.test.spec.CustomConfigSpec
+import com.exactpro.th2.test.spec.RabbitMqSpec
+import com.exactpro.th2.test.spec.RabbitMqSpec.Companion.EVENTS_PIN_NAME
+import com.exactpro.th2.test.spec.filter
+import com.exactpro.th2.test.spec.message
+import com.exactpro.th2.test.spec.pin
+import com.exactpro.th2.test.spec.pins
+import com.exactpro.th2.test.spec.publishers
+import com.exactpro.th2.test.spec.subscribers
+import mu.KotlinLogging
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import strikt.api.Assertion
+import strikt.api.expectThat
+import strikt.assertions.all
+import strikt.assertions.hasSize
+import strikt.assertions.isEmpty
+import strikt.assertions.isEqualTo
+import strikt.assertions.isFalse
+import strikt.assertions.matches
+import strikt.assertions.withElementAt
+import java.time.Duration.ofSeconds
+import java.time.Instant
+import java.util.concurrent.ConcurrentLinkedDeque
+import kotlin.test.assertNotNull
+
+@IntegrationTest
+@Th2IntegrationTest
+class ApplicationIntegrationTest {
+    @JvmField
+    @Suppress("unused")
+    internal val customConfig = CustomConfigSpec.fromString(
+        """
+        {
+            "host": "127.0.0.1",
+            "port": 8080,
+            "sessionAlias": "some_api",
+            "validateCertificates": false,
+            "sessionAlias": "test-session-alias",
+            "useTransport": true
+        }
+        """.trimIndent()
+    )
+
+    @JvmField
+    @Suppress("unused")
+    internal val mq = RabbitMqSpec.create()
+        .pins {
+            subscribers {
+                pin("sub") {
+                    attributes(INPUT_QUEUE_TRANSPORT_ATTRIBUTE, TRANSPORT_GROUP_ATTRIBUTE)
+                    filter {
+                        message {
+                            field("test") shouldBeEqualTo "a"
+                        }
+                    }
+                }
+            }
+
+            publishers {
+                pin("pub") {
+                    attributes(TRANSPORT_GROUP_ATTRIBUTE)
+                }
+            }
+        }
+
+    private val resources = ConcurrentLinkedDeque<Pair<String, () -> Unit>>()
```

Comment:
There is a cleanup extension in th2 junit extension library. Please read the doc: https://github.com/th2-net/junit-jupiter-integration/blob/master/README.md